cpp/include/cudf/rolling.hpp

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+
+#include <memory>
+
+namespace cudf {
+namespace experimental {
+
+/** 
+ * @brief Rolling window aggregation operations
+ */
+enum rolling_operator {
+  SUM,       ///< Computes the sum of all values in the window
+  MIN,       ///< Computes minimum value in the window
+  MAX,       ///< Computes maximum value in the window
+  AVG,       ///< Computes arithmetic mean of all values in the window
```

Comment:
We use `MEAN` in groupby (and I think other aggs). 
```suggestion
  MEAN,       ///< Computes arithmetic mean of all values in the window
```

Reply:
Not in rolling, apparently! (this was a copy-paste from the old API. But I agree MEAN is NICE, will change.